src/main/java/com/endava/web/application/domain/service/EmployeeService.java

```diff
@@ -0,0 +1,80 @@
+package com.endava.web.application.domain.service;
+
+import com.endava.web.application.domain.dao.DepartmentRepository;
+import com.endava.web.application.domain.dao.EmployeeRepository;
+import com.endava.web.application.domain.exception.exceptions.DepartmentConstraintsException;
+import com.endava.web.application.domain.exception.exceptions.DepartmentDoesNotExistsException;
+import com.endava.web.application.domain.exception.exceptions.NoSuchDepartmentException;
+import com.endava.web.application.domain.exception.exceptions.NoSuchEmployeeException;
+import com.endava.web.application.domain.model.Employee;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository repositoryDepartment;
+
+    public List<Employee> getAllEmployees() {
+        return employeeRepository.findAll();
+    }
+
+    public Employee getEmployee(int employeeId) {
+        Optional<Employee> employee = employeeRepository.findById(employeeId);
+        return employee.orElseThrow(() -> new NoSuchEmployeeException(
+                "EMPLOYEE WITH ID " + employeeId + " DOES NOT EXISTS"));
+    }
+
+    public Employee saveEmployee(Employee employee) {
+        checkEmployeeNameConstraints(employee);
+        if(!repositoryDepartment.findDepartmentById(employee.getDepartment().getId()).isPresent()){
```

Comment:
is it a requirement that department should already exist at this point ?
your employee has department as member

Reply:
I corrected it. We can have employee without department, but we cannot have two departments with the same name. Done.